rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -95,6 +110,17 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
               case Some(joins) => Applicative[F].pure(joins)
             }
     } yield (res)
+
+  private def removeIndex[E](col: Vector[E], index: Int): F[Vector[E]] =
+    if (col.isDefinedAt(index)) {
+      val (l1, l2) = col splitAt index
+      (l1 ++ (l2 drop 1)).pure[F]
```

Comment:
`2 drop 1` tail?

Reply:
?

Reply:
`(l1 ++ (l2.tail)).pure[F]` ?

Reply:
No, this isn't going to work with an empty list